### DISCUSSION TITLE
autobackup/image like other servers...

### DISCUSSION THREAD

**Discussion by pkarza** [immich-app/immich#9838] [Q&A]
I am migrating to immich (got it working and using tailscale for remote)  - but there is a feature that is missing or I have missed  - 
I hope I am allowed to use the name of the other server I have - a  Monument 1 - from getmonument.
I like immich but do not feel secure because there is not mirroring of the images/videos and database.
As a matter of fact - I wanted to do this with rsync - but I can't find where my images and videos are (in entirety) .  I have a library directory in my immich-app folder - but it only shows 70 gb used - but immich is showing 90 gb used of the 256 gb drive.

If this feature is missing (or I missed it ) it should be implemented.  2 mirrored drives would be a plus and add another level of security.

Any help greatly appreciated!!

**Comment by mmomjian** [comment]
The images and videos are stored in `UPLOAD_LOCATION`, wherever this path is set to in your `.env` file. The data storage usage on Immich considers your entire system drive, so that means you have 20GB of non-Immich data on your PC. The immich-app is probably what you were looking for.

Mirroring data is not something that will be handled by Immich, you can handle this on the system level.

**Reply by pkarza** [reply]
Thank you for the reply - I left the variable to UPLOAD_LOCATION= ./library - I thought that would imply a hidden directory.  

My structure is:
~/immich-app
      - library
           -  encoded-video
           - thumbs
           - upload
                - 7e5b880d-ba3f-4f10-8f88-bcc631963a82
                     - many sub directories...
                - 5539a385-d928-4c72-a1c3-f34ed5b43331
                      - many sub directories
           - library ( <--- nothing in here ?? what is it's purpose?)

So - if I were to set up a cron job to rsync the immich-app directory - everything would be copied?  would that include the db files - or should I just include the entire drive and set the cron job up to --ignore-existing ?

This really should be part of the program as an addon ...  The documentation is a little weak on what to backup - and as an additional suggestion there should be a faq that would explain things like this and tailscale to those who do not understand these things.

Thanks for the patience of my questions - I couldn't find definitive answers elsewhere.

Phil       
     
      -

**Reply by pkarza** [reply]
By the way - this is much better than what I was using before.  Happened upon it by accident and love it.

**Reply by mmomjian** [reply]
The folder structure you posted contains all the images. It does not contain the database.

The backup structure and process is outlined in quite some detail here: https://immich.app/docs/administration/backup-and-restore/
